providers/prefab/pkg/provider_test.go

```diff
@@ -0,0 +1,390 @@
+		require.Equal(t, 1.2, resolution.Value)
+	})
+
+	evalCtx := of.NewEvaluationContext(
+		"",
+		map[string]interface{}{},
+	)
+	value, err := ofClient.FloatValue(context.Background(), "sample_double", 1.2, evalCtx)
+	require.Nil(t, err)
+	require.Equal(t, 12.12, value)
+
+}
+
+func TestIntEvaluation(t *testing.T) {
+
+	flattenedContext := map[string]interface{}{}
+
+	resolution := provider.IntEvaluation(context.Background(), "sample_int", 1, flattenedContext)
+	if resolution.Value != 123 {
+		t.Fatalf("Expected one of the variant payloads")
+	}
+
+	t.Run("evalCtx empty", func(t *testing.T) {
+		resolution := provider.IntEvaluation(context.Background(), "non-existing", 1, nil)
+		require.Equal(t, int64(1), resolution.Value)
+	})
+
+	evalCtx := of.NewEvaluationContext(
+		"",
+		map[string]interface{}{},
+	)
+	value, err := ofClient.IntValue(context.Background(), "sample_int", 1, evalCtx)
+	require.Nil(t, err)
+	require.Equal(t, int64(123), value)
+
+}
+
+func TestStringEvaluation(t *testing.T) {
+
+	flattenedContext := map[string]interface{}{}
+
+	resolution := provider.StringEvaluation(context.Background(), "sample", "default", flattenedContext)
+	if resolution.Value != "test sample value" {
+		t.Fatalf("Expected one of the variant payloads")
+	}
+
+	t.Run("nil evalCtx", func(t *testing.T) {
+		resolution := provider.StringEvaluation(context.Background(), "non-existing", "default", nil)
+		require.Equal(t, "default", resolution.Value)
+	})
+
+	evalCtx := of.NewEvaluationContext(
+		"",
+		map[string]interface{}{},
+	)
+	value, err := ofClient.StringValue(context.Background(), "sample", "default", evalCtx)
+	require.Nil(t, err)
+	require.Equal(t, "test sample value", value)
+
+}
+
+// TODO test and enable when json/yaml parsing is implemented
+func TestObjectEvaluation(t *testing.T) {
+
+	flattenedContext := map[string]interface{}{}
+
+	t.Run("example.nested.path", func(t *testing.T) {
+		resolution := provider.ObjectEvaluation(context.Background(), "example.nested.path", "default", flattenedContext)
+		require.Equal(t, "hello", resolution.Value)
+	})
+
+	evalCtx := of.NewEvaluationContext(
+		"",
+		map[string]interface{}{},
+	)
+
+	t.Run("example.nested.path", func(t *testing.T) {
+		value, err := ofClient.ObjectValueDetails(context.Background(), "example.nested.path", "default", evalCtx)
+		require.Equal(t, "hello", value.Value)
+		require.Nil(t, err)
+	})
+
+	t.Run("sample_list", func(t *testing.T) {
+		value, err := ofClient.ObjectValueDetails(context.Background(), "sample_list", []string{"a2", "b2"}, evalCtx)
+		require.Equal(t, []string{"a", "b"}, value.Value)
+		require.Nil(t, err)
+	})
+
+	// TODO
+	// t.Run("sample_json", func(t *testing.T) {
+	// 	value, err := ofClient.ObjectValueDetails(context.Background(), "sample_json", map[string]interface{}{
+	// 		"nested": "value",
+	// 	}, evalCtx)
+	// 	require.Equal(t, []string{"a", "b"}, value.Value)
+	// 	require.Nil(t, err)
+	// })
```

Comment:
Should this be removed?

Reply:
Oh I guess this is related to https://github.com/open-feature/go-sdk-contrib/pull/575/files#diff-cda080e5d119928854541dc260e063a153933ee2c950795929524c75e1655eb0R74

Reply:
yes, can leave it for now, and will track to see when can be enhanced with the additional tests.